gatsby/content/blog/2021/11/2021-11-XX-sydent-typing.mdx

```diff
@@ -0,0 +1,803 @@
+---
+  date: '2021-11-08'
+  title: 'Type coverage for Sydent: lessons learnt'
+  categories:
+    - Thoughts
+    - Tech
+  author: David Robertson
+---
+
+[Sydent](https://github.com/matrix-org/sydent) is the reference Matrix Identity server. It provides a [lookup service](https://matrix.org/docs/spec/identity_service/r0.3.0), so that you can find a Matrix user given only their email address or phone number. We recently worked on [improving Sydent's typing coverage](https://github.com/matrix-org/sydent/issues/414); it now passes `mypy --strict` with an overall precision of 94%! 
+
+![Coverage as measured by mypy. Precision and the number of typed expressions increase over the Summer and Autumn of 2021.](/blog/img/TWGXKTFpjYxlbnHRkZFAzljQ.png)
+
+We can see our progress on the graph above. The grey region covers our two-week sprint towards improving coverage. The earliest data point is from just before our earlier efforts to improve typing this summer.
+
+"Typing" refers to the process of annotating Python source code, so that automated tools can reason about the types of expressions that flow through a program. These annotations can be automatically checked by a tool like [`mypy`]
+(https://mypy.readthedocs.io/en/stable/)—a great way to spot bugs before you ship them.
```

Comment:
think you need a space to avoid confusing the markdown parser

```suggestion
(https://mypy.readthedocs.io/en/stable/) — a great way to spot bugs before you ship them.
```